src/services/slack.ts

```diff
@@ -52,25 +53,44 @@ export class SlackConnector implements ISlackConnector {
     }
     return {};
   }
-  parseSelection(stateValues: any): any {
+
+  async parseSelection(
+    stateValues: any,
+    isAdmin: boolean,
+    repoBranchesRepository: RepoBranchesRepository
+  ): Promise<any> {
```

Comment:
nit: if possible, can we make the return value more specific? But if that's complicated, totally not-blocking!

Reply:
I'm open to that! However, I'm not sure what approach to take in this situation. `parseSelection` always returns an object called `values` as a Promise, but the attributes of `values` are inconsistent so it's not strongly typed itself. Ex: `values` always has a `repo_option` property but only sometimes a `hash_option` or `deploy_option`.
  I believe(?) specifying the return type further than Promise<object> would require a refactor of the function and some checks added elsewhere, but please correct me if you have something else in mind! If not, happy to do this in the follow up ticket to this

Reply:
Yeah, non-blocking! We can always do this later! Seems like  a larger time commitment :)